Distribution/Server/Features/UserSignup.hs

```diff
@@ -419,7 +403,9 @@ userSignupFeature ServerEnv{serverBaseURI, serverCron}
     --
 
     nonceInPath :: MonadPlus m => DynamicPath -> m Nonce
-    nonceInPath dpath = maybe mzero return (Nonce . BS.pack <$> lookup "nonce" dpath)
+    nonceInPath dpath =
+        do raw <- maybe mzero return (lookup "nonce" dpath)
+           parseNonceM raw
```

Comment:
This `parseNonceM` needs to be inside the `maybe mzero return` because `parseNonceM` returns fail, but we want mzero not fail. mzero will backtrack but fail will give a 5xx failure.

I can fix this one.